### ISSUE TITLE
Storage prefix doesn't support "" any longer.

### ISSUE THREAD

**Issue by gnarl** [NetApp/trident#430]
**Describe the bug**
With Trident 20.07.0 the use of "" or "-" in the storage prefix results in an error.

**Environment**
Provide accurate information about the environment to help us reproduce the issue.

- Trident version: 20.07.0
- Trident installation flags used: any

**To Reproduce**
Add a storage backend with a storage prefix of "" or that includes "-".

**Expected behavior**
I should be able to use "" or "-" in the storage prefix.

**Comment by lopf** [comment]
We just updated 20.01.1 to 20.07.0 and ran into this as well. Our prefix includes a "-".

```
time="2020-08-11T08:19:05Z" level=error msg="Could not initialize storage driver." error="error validating ontap-nas driver: driver validation failed: storage prefix may only contain letters/digits/underscore and must begin with letter/underscore"
time="2020-08-11T08:19:05Z" level=warning msg="Cannot terminate an uninitialized backend." backend= backendUUID= driver=ontap-nas state=failed
time="2020-08-11T08:19:05Z" level=error msg="problem initializing storage driver 'ontap-nas': error validating ontap-nas driver: driver validation failed: storage prefix may only contain letters/digits/underscore and must begin with letter/underscore" backend= handler=UpdateBackend
```

Now we're stuck because updating it is not supported:

```
updating the storage prefix isn't currently supported (400 Bad Request)
command terminated with exit code 1
```

**Comment by gnarl** [comment]
@lopf we are working on providing a patch release to fix this issue. I'll also update this issue once the commit with the fix is merged.

**Comment by sevenval-admins** [comment]
@gnarl Is there an approximate date for the patch release? We are eagerly waiting for it so that we can continue with the next clusters.

**Comment by gnarl** [comment]
@sevenval-admins, I can't give you a solid date but we are currently working on bug fixes and also need to qualify K8S 1.19 for a 20.07.1 patch release. The storage prefix fix is already committed [here](https://github.com/NetApp-openstack-dev/trident/commit/7e059eff4f079a50bc217fa102815f5453cdac81).

**Comment by gnarl** [comment]
This is fixed with the [Trident v20.07.1](https://github.com/NetApp/trident/releases/tag/v20.07.1) release.
